coral/src/app/features/topics/browse/BrowseTopics.test.tsx

```diff
@@ -513,34 +374,42 @@ describe("BrowseTopics.tsx", () => {
     });
 
     it("changes active selected option when user selects `TEST_TEAM_02`", async () => {
-      await waitForElementToBeRemoved(screen.getByText("Loading..."));
       const select = screen.getByRole("combobox", {
         name: filterByTeamLabel,
       });
-      const option = within(select).getByRole("option", {
+
+      const option = await screen.findByRole("option", {
         name: "TEST_TEAM_02",
       });
+
       expect(select).toHaveValue("f5ed03b4-c0da-4b18-a534-c7e9a13d1342");
 
       await userEvent.selectOptions(select, option);
 
       expect(select).toHaveValue("TEST_TEAM_02");
     });
 
-    it("fetches new data when user selects `TEST_TEAM_02`", async () => {
-      const spyGet = jest.spyOn(api, "get");
-      const url = getUrlWithParams({
-        route: "getTopics",
-        teamName: "TEST_TEAM_02",
+    it("shows an information that the list is updated after user selected a team", async () => {
+      const select = screen.getByRole("combobox", {
+        name: filterByTeamLabel,
       });
-      const getAllTopics = () =>
-        within(
-          screen.getByRole("table", { name: /Topics overview/ })
-        ).getAllByRole("rowheader");
-      await waitForElementToBeRemoved(screen.getByText("Loading..."));
+      const option = within(select).getByRole("option", {
+        name: "TEST_TEAM_02",
+      });
+      expect(select).toHaveValue("f5ed03b4-c0da-4b18-a534-c7e9a13d1342");
 
-      expect(getAllTopics()).toHaveLength(10);
+      // I'm not happy with that, but it' s the only way I found
+      // to make sure there's a information about the list being
+      // updated rendered for the user. awaiting for the userEvent
+      // will end in a state where the new data has already
+      // arrived and is rendered, to the updating info will be gone
+      userEvent.selectOptions(select, option);
```

Comment:
Could we add a `setTimeout` for the mock that returns the relevant data? then we can reliably be sure that not awaiting this means there will be a 500ms window in which to assert the loading state?

Reply:
Hm. I honestly would rather test the "Filtering list..." behavior in a different way 🤔  Relying on what I did is already wacky (more than I like), but having to add timeouts to a test usually a bad sign (well, on this test level). 

Since all these user-infos ("isLoading", "isPreviousData" etc) are making use of the query, I thought we could have one test block where the query is mocked and we check for the different views dependent on the returned values from the query. 

That would fulfil our need - making sure users get certain information about updates when loading data. 

Testing them in the setup like right now does implicitly test that react-query is working, which is unneeded. It does not add additional value for our test case. 

🤔

Reply:
That's true, a query mock would be better. Anyway, I will merge this, as the objective of the PR has been reached :o